oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/NodeDocument.java

```diff
@@ -2551,7 +2550,7 @@ private static final class ValueComparator implements
 
         static final Comparator<Entry<Revision, String>> REVERSE = Collections.reverseOrder(INSTANCE);
 
-        private static final Ordering<String> STRING_ORDERING = Ordering.natural().nullsFirst();
+        private static final Comparator<String> STRING_ORDERING = Comparator.nullsLast(Comparator.naturalOrder());
```

Comment:
shouldn't this be "nullsFist"?

Reply:
Thanks @reschke for catching this.

Reply:
fixed in https://github.com/apache/jackrabbit-oak/pull/2314/commits/4a32eafde4cfca3452207d34636c8c08e72a87c8